src/core/int128.d

```diff
@@ -0,0 +1,748 @@
+/* 128 bit integer arithmetic.
+ *
+ * Not optimized for speed.
+ *
+ * Copyright: Copyright D Language Foundation 2022.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   Walter Bright
+ * Source:    $(DRUNTIMESRC core/_int128.d)
+ */
+
+module core.int128;
+
+nothrow:
+@safe:
+@nogc:
+
+alias I = long;
+alias U = ulong;
+enum Ubits = U.sizeof * 8;
+
+struct Cent
+{
+    U lo;      // low 64 bits
+    U hi;      // high 64 bits
+}
+
+enum One = Cent(1);
+enum Zero = Cent();
+enum MinusOne = neg(One);
+
+/*****************************
+ * Test against 0
+ * Params:
+ *      c = Cent to test
+ * Returns:
+ *      true if != 0
+ */
+pure
+bool tst(Cent c)
+{
+    return c.hi || c.lo;
+}
```

Comment:
Every single use of this function is `!tst(c)`, so it'd be much better just naming it `bool isZero(Cent c)` and inverting the condition.

Reply:
I'm reminded of `isEmpty()` for ranges, and every time I use it I have to stop and think a moment.

Reply:
True means non-zero and False means zero. Anything that flips that about, like isZero, is inherently confusing to me.

`tst` is not inherently confusing. It's been in wide use for at least 50 years now to mean !=0, probably more, and I've never heard of *anyone* being confused by it. I was a newbie once, too, and don't recall any confusion with it.

I have used inverted logic (`isZero` is inverted logic), and the only reason it is used (NAND gates, NOR gates) is because one less transistor is required. NAND logic is inherently confusing, and hard to get used to. I'd design a circuit using positive logic, and then mechanistically convert it to inverted logic using DeMorgan's Theorem.